### ISSUE TITLE
decimal.InvalidOperation error for fixed delivery futures' funding data on BinanceDelivery

### ISSUE THREAD

**Issue by nirvana-msu** [bmoscon/cryptofeed#672]
**Describe the bug**
For contracts with fixed delivery, BinanceDelivery API returns funding rate as an empty string:
https://binance-docs.github.io/apidocs/delivery/en/#mark-price-stream
```
    "r":"",                 // funding rate for perpetual symbol, "" will be shown for delivery symbol
    "T":0                   // next funding time for perpetual symbol, 0 will be shown for delivery symbol
```
This results in an exception when empty string is passed to `Decimal`.

**To Reproduce**
```python
from cryptofeed import FeedHandler
from cryptofeed.defines import BINANCE_DELIVERY, FUNDING
from cryptofeed.types import Funding

async def funding(funding: Funding, receipt_timestamp: float):
    pass

fh = FeedHandler()
fh.add_feed(
    BINANCE_DELIVERY, symbols=['BTC-USD-21Z31'], channels=[FUNDING], callbacks={FUNDING: funding},
    depth_interval='500ms',
)
fh.run()
```
```
2021-10-03 01:55:06,574 : ERROR : BINANCE_DELIVERY.ws.1: encountered an exception, reconnecting in 1.0 seconds
Traceback (most recent call last):
  File "/home/<...>/venv/lib/python3.9/site-packages/cryptofeed/connection_handler.py", line 68, in _create_connection
    await self._handler(connection, self.handler)
  File "/home/<...>/venv/lib/python3.9/site-packages/cryptofeed/connection_handler.py", line 117, in _handler
    await handler(message, connection, self.conn.last_message)
  File "/home/<...>/venv/lib/python3.9/site-packages/cryptofeed/exchanges/binance_delivery.py", line 156, in message_handler
    await self._funding(msg, timestamp)
  File "/home/<...>/venv/lib/python3.9/site-packages/cryptofeed/exchanges/binance.py", line 401, in _funding
    Decimal(msg['r']),
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]
```

**Expected behavior**
Funding rate should be set to `None`.

While we're on the subject, "next funding time" does not apply to fixed-maturity contracts either. Binance API returns "0" as the time - and while there are no exceptions, I would argue that it would be more intuitive for cryptofeed to set "next funding time" for non-perpetual contracts to `None`, rather than `0`.

Same apples to `BinanceFutures` exchange. While most contracts there are perpetual, there are two contracts with fixed maturity, e.g. here's how to subscribe to one of them:
```python
fh.add_feed(
    BINANCE_FUTURES, symbols=['BTC-USDT-21Z31'], channels=[FUNDING], callbacks={FUNDING: funding},
    depth_interval='500ms',
)
```
`BinanceFutures`, unlike `BinanceDelivery`, returns funding rate as `"0.00000000"` and not `""` so there is no exception, however it would still make more sense to set both funding rate and next funding time to `None` as these terms do not apply for such contracts.

To sum up, I would expect funding rate and next funding time to be set to `None` for both `BinanceFutures` and `BinanceDelivery` for non-perpetual futures. We could simply check if symbol ends with `PERP`.

**Cryptofeed Version**
Latest release, 2.0.1

**Comment by nirvana-msu** [comment]
@bmoscon thanks. Added one comment to PR - we should not be replacing all zero funding rates with `None`, as it may well be a valid rate.
